File: Data Structures & Algorithms/balanced-binary-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public boolean isBalanced(TreeNode root) {
        if(root == null){
            return true;
        }else{
            System.out.println((height(root.left)) +" "+(height(root.right)));
            return (Math.abs((height(root.left) - height(root.right))) <=1 ) && 
            isBalanced(root.left) && isBalanced(root.right);
        }
    }

    public int height(TreeNode node){
        if(node == null){
            return 0;
        }else if(node.left == null && node.right == null){
            return 1;
        }else{
            return Math.max(height(node.left) , height(node.right))+1;
        }
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        outputList = []
        bfsQueue = deque()
        bfsQueue.append(root)
        while bfsQueue:
            size = len(bfsQueue)
            levelArr = []
            while size > 0:
                popEle = bfsQueue.popleft()
                size -= 1
                if popEle:
                    bfsQueue.append(popEle.left)
                    bfsQueue.append(popEle.right)
                    levelArr.append(popEle.val)

            if levelArr:
                outputList.append(levelArr)

        return outputList

            

        